test/e2e/generator/generate.go

```diff
@@ -196,6 +206,14 @@ func generateNode(r *rand.Rand, mode e2e.Mode, startAt int64, forceArchive bool)
 		node.SnapshotInterval = 3
 	}
 
+	if node.Mode == "validator" && node.PrivvalProtocol == "file" {
+		misbehaviorHeight := startAt + 20
```

Comment:
20 is too high, I think. In the trivial case, just starting four validators without anything fancy, it will have proceeded to tests and possibly even shut down the network by the time height 20 is reached.

Reply:
10 then?

Reply:
Yeah, maybe. This gets really hard, because it'll depend on so many factors: when it joins the validator set, how long it takes to sync up (i.e. whether it uses fast sync/state sync or not), if there are any perturbations scheduled, whether it's the last network node being started, and so on. It'd be better to just trigger it on demand.

Reply:
Yeah do you think we could introduce a misbehave phase then. As a bit of a halfway solution you could make misbehave calculate the height of the last misbehavior and then wait until that block is reached before beginning the perturb phase.

Reply:
The thing about doing it on demand is also figuring out the height when the evidence is expected to have be created

Reply:
>Yeah do you think we could introduce a misbehave phase then.

Yes, but it's probably going to be too much work right now. If we're able to come up with a reasonably reliable scheme (i.e. such that we don't get lots of nightly testnet flake because of this) then that'll do for now.

>As a bit of a halfway solution you could make misbehave calculate the height of the last misbehavior and then wait until that block is reached before beginning the perturb phase.

I think this is a good idea in any case, but maybe not as a separate stage (since we won't be able to trigger it with `runner misbehave`), just as a simple `waitFor` in the default run sequence. This should probably happen before we go on to perturbations.

This reminds me, the test should also check that there's no misbehavior scheduled beyond the chain head (i.e. misbehavior that never happened).

>The thing about doing it on demand is also figuring out the height when the evidence is expected to have be created

It'd be sufficient to simply check that it happened, exactly when isn't really that interesting.

Reply:
I guess the other this is that with having a constant it means that most misbehaviors happen at the same time

Reply:
> beyond the chain head 

I guess this won't happen if we call a waitFor function till the end of the last misbehavior

Reply:
True, for validators that start at the initial height. Maybe we need to introduce some randomness or otherwise make sure they're staggered.